### PR TITLE
Add option to create key pair without saving onto firewall on "Create Internal Certificate"

### PR DESCRIPTION
**Pull request by nhirokinet**
This is a proposal for #3272 . This enables creating a private only for the device who opens admin web page.

This is not elegant for the point that secret key is sent back to the OPNsense, which produces double traffic, but on situation that OPNsense does not want to save the secret key, I thought significant improvement against this is hard.

### CODE REVIEW

**Comment by AdSchellevis** [comment]
I think we can simplify this a lot by just returning the private key data when generated and let the user save the certificate to a file (start with a text area containing the key).

unset the private key here when requested:
https://github.com/opnsense/core/blob/f685abcc9c91805e9f5dad35f8f30bc4bb6c585f/src/www/system_certmanager.php#L738-L740

Fall through after post, ignoring the redirect here
https://github.com/opnsense/core/blob/f685abcc9c91805e9f5dad35f8f30bc4bb6c585f/src/www/system_certmanager.php#L751-L756

Then render the page with the private key, you can probably use `$act` for flow control.

It might still not be the prettiest solution in the world, and to be honest I rather see cleanups to the code first before adding more logic, but it could lead to an acceptable solution.

**Comment by nhirokinet** [comment]
Thank you for comment, I tried to improve my code accoding to your advice.

For newly written HTML I referred some HTML in other places of OPNsense, but I'm not sure whether this is expected in OPNsense.

File: src/www/system_certmanager.php [hunk 1]
```diff
@@ -1087,9 +1108,41 @@ function removeRowAltNm() {
         $("#certmethod").change();
     }
   });
+
+  function refresh_download_link(jquery_key, content, filename) {
+      if (navigator.userAgent.indexOf('MSIE ') !== -1 || navigator.userAgent.indexOf('Trident') !== -1) {
+          // IE11 support; they do not have <a download="">
```
Comment (AdSchellevis) [inline src/www/system_certmanager.php]: I'm not sure we should go through all that trouble trying to support IE11, what if we just make sure the user can copy-paste the content and optionally save to file if supported by the browser?

Comment (nhirokinet) [reply src/www/system_certmanager.php]: Okay, I will remove download link if unsupported.

**Comment by nhirokinet** [comment]
Edited so that check download attribute is available, and if not remove the link.

**Comment by AdSchellevis** [comment]
@nhirokinet cleanups in https://github.com/opnsense/core/commit/fa7e7d878a05b7bc09281974e142420bdeb906f5

**Comment by AdSchellevis** [comment]
thanks!

**Pull request merged**